tests/test_boto_helpers.py

```diff
@@ -200,6 +202,43 @@ def test_get_temporary_credentials_assume_role():
     assert tuple(creds.values()) == (access_key, secret_key, token)
 
 
+def test_get_temporary_credentials_assume_role_with_headers():
```

Comment:
I believe there is two unit test scenarios not accounted for:
1. source_account is provided but no source_arn
2. source_arn is provided but no source_account